allennlp/training/trainer.py

```diff
@@ -140,6 +140,30 @@ def __call__(
         raise NotImplementedError
 
 
+class CustomlySizedEpochDataLoader:
```

Comment:
Can you move this inside `dataloader.py`?  Also, maybe name it `CustomEpochSizeDataLoader`?